src/components/implementation/tests/patina_chan_bench/patina_chan_bench.c

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2020, Bite Ye, Runyu Pan and Gabriel Parmer, GWU, gparmer@gwu.edu.
+ *
+ * This uses a two clause BSD License.
+ */
+
+#include <llprint.h>
+#include <sched.h>
+
+#include <patina.h>
+#include <perfdata.h>
+
+#define COLD_CACHE
+#ifdef COLD_CACHE
+#define cache_flush() __cache_flush()
+#define COLD_OFFSET 1
+#define COLD_INDEX 0
+#else
+#define cache_flush()
+#define COLD_OFFSET 0
+#define COLD_INDEX -1
+#endif
+
+#define CACHE_SIZE 512 * 1024
+#define CACHE_LINE_SIZE 32
+
+#undef PATINA_CHAN_TRACE_DEBUG
+#ifdef PATINA_CHAN_TRACE_DEBUG
+#define debug(format, ...) printc(format, ##__VA_ARGS__)
+#else
+#define debug(format, ...)
+#endif
+
+/* One low-priority thread and one high-priority thread contends on the lock */
+#ifdef COLD_CACHE
+#define ITERATION 10 * 10
+#else
+#define ITERATION 10 * 1000
+#endif
+#undef PRINT_ALL
+
+/* Two options are available: Sender at low/high prio, data words 4 */
+#define DATA_WORDS 2
+
+thdid_t chan_reader = 0, chan_writer = 0;
+
+typedef unsigned int cycles_32_t;
+
+volatile cycles_32_t tmp[DATA_WORDS] = {
+  0,
+};
+volatile cycles_32_t ts1[DATA_WORDS] = {
+  0,
+};
+volatile cycles_32_t ts2[DATA_WORDS] = {
+  0,
+};
+volatile cycles_32_t ts3[DATA_WORDS] = {
+  0,
+};
+
+struct perfdata perf1, perf2, perf3;
+cycles_t        result1[ITERATION] = {
+  0,
+};
+cycles_t result2[ITERATION] = {
+  0,
+};
+cycles_t result3[ITERATION] = {
+  0,
+};
+
+patina_chan_t   cid;
+patina_chan_t   cid2;
+patina_chan_r_t rid;
+patina_chan_r_t rid2;
+patina_chan_s_t sid;
+patina_chan_s_t sid2;
+
+volatile char pool[CACHE_SIZE * 4] = {
+  0,
+};
+
+void
+__cache_flush()
+{
+	int agg = 1;
+	for (int i = 0; i < CACHE_SIZE * 4; i += CACHE_LINE_SIZE) {
+		pool[i] += agg;
+		agg = pool[i];
+	}
+}
+
+/***
+ * The two threads reciprocally sends and receives.
+ */
+void
+chan_reader_thd(void *d)
+{
+	/* Never stops running; writer controls how many iters to run. */
+	while (1) {
+		debug("r1,");
```

Comment:
Even worse, the debug statements make this hard to read! I know you didn't write them originally.